app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -510,16 +510,19 @@ class DefaultSessionControlController(
     }
 
     override fun handleShowWallpapersOnboardingDialog(state: WallpaperState): Boolean {
-        if (state.availableWallpapers.all { it.thumbnailFileState == Wallpaper.ImageFileState.Downloaded } &&
-            state.availableWallpapers.size >= WallpaperOnboardingDialogFragment.THUMBNAILS_COUNT
-        ) {
-            navController.nav(
-                R.id.homeFragment,
-                HomeFragmentDirections.actionGlobalWallpaperOnboardingDialog(),
-            )
-            return true
+        return state.availableWallpapers.filter { wallpaper ->
+            wallpaper.thumbnailFileState == Wallpaper.ImageFileState.Downloaded
+        }.size.let { downloadedCount ->
+            // We only display the dialog if enough thumbnails have been downloaded for it.
+            downloadedCount >= WallpaperOnboardingDialogFragment.THUMBNAILS_SELECTION_COUNT
+        }.also { isThumbnailsDownloaded ->
+            if (isThumbnailsDownloaded) {
```

Comment:
```suggestion
            if (downloadedCount >= THUMBNAILS_SELECTION_COUNT) {
```
nit: "Less is more" or so the saying goes. Since you added a comment describing the conditional, I don't think there's really a point in naming the conditional to `isThumbnailsDownloaded`. I would suggest shortening the amount of code that is necessary here.

Also for your imports, you could import `import org.mozilla.fenix.onboarding.WallpaperOnboardingDialogFragment.THUMBNAILS_SELECTION_COUNT` to be less verbose.

Reply:
Hey! Thanks for the reminder. I was actually waiting for some feedback from @MatthewTighe or you before landing. And assuming there might be more changes, didn't squash into one, so the trackrecord would be there.

Regarding the shortening, I am not sure I follow you line of though. Are you suggesting something like this? We mess up the returning type this way, it becomes unit, not boolean

```
return state.availableWallpapers.filter { wallpaper ->
            wallpaper.thumbnailFileState == Wallpaper.ImageFileState.Downloaded
        }.size.let { downloadedCount ->
            // We only display the dialog if enough thumbnails have been downloaded for it.
            if (downloadedCount >= WallpaperOnboardingDialogFragment.THUMBNAILS_SELECTION_COUNT) {
                navController.nav(
                    R.id.homeFragment,
                    HomeFragmentDirections.actionGlobalWallpaperOnboardingDialog(),
                )
            }
        }
```

Reply:
Yes I was suggesting what you have provided. I do see the boolean now so, I understand that we couldn't shorten this. Apologizes.

Reply:
Oh, okay, gotcha. Changed to `showOnboarding`, as well addressing the import as well.

Reply:
👍 Looks good. Thanks!